src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VCFComparator.java

```diff
@@ -0,0 +1,887 @@
+
+        if (warningsOutput != null) {
+            outputStream = new PrintStream(warningsOutput.getOutputStream());
+        }
+    }
+
+    @Override
+    public void apply(final List<VariantContext> variantContexts, final ReferenceContext referenceContext, final List<ReadsContext> readsContexts) {
+        if (isSingleSample && !variantContexts.get(0).getAlleles().contains(Allele.NON_REF_ALLELE)) {
+            throw new UserException.BadInput("Single-sample mode expects two GVCFs with <NON_REF> data for comparison");
+        }
+
+        if (variantContexts.size() == 1) {
+            final VariantContext vc = variantContexts.get(0);
+            //TODO: if there's a variant that overlaps the requested interval the start may not be in the requested intervals
+            if (!muteDiffs) {
+                throwOrWarn(new UserException("Unmatched variant in " + vc.getSource() + " at position " + vc.getContig() + ":" + vc.getStart()));
+            //low coverage sites will have QUAL for singleton hom-vars boosted because of GQ0 hom-refs
+            } else if (hasGoodEvidence(vc)) {
+                throwOrWarn(new UserException("Unmatched variant in " + vc.getSource() + " at position " + vc.getContig() + ":" + vc.getStart()));
+            }
+        }
+
+        //here we have all variants overlapping a given position
+        //if there's no actual for the expected (at the same position), see if the expected has an overlapping deletion
+        for (final VariantContext vc : variantContexts) {
+            if (vc.getSource().equals("actual")) {  //there may be more expected than actual variants, but only compare once
+                continue;  //TODO: this means we don't report cases where actual has multiple reference blocks spanning expected as unmatched
+            }
+            //vc is guaranteed to be from expected
+
+            final List<VariantContext> matches = variantContexts.stream().filter(v -> v.getStart() == vc.getStart()).collect(Collectors.toList());
+            //matches includes vc
+            //TODO: matches.size() could be 4 or more, depending on ploidy
+            if (matches.size() == 1 ) {
+                if (isHighQuality(vc) && hasGoodEvidence(vc)
+                        && !vc.getGenotype(0).getAlleles().contains(Allele.SPAN_DEL)){
+                    throwOrWarn(new UserException("Apparent unmatched high quality variant in " + vc.getSource() + " at " + vc.getContig() + ":" + vc.getStart()));
+                } else {
+                    return;
+                }
+            } else {
+                if ((!isHighQuality(vc) || (isSingleSample && !GATKVariantContextUtils.genotypeHasConcreteAlt(vc.getGenotype(0).getAlleles())))
+                && ignoreHomRefAttributes) { //TODO: check * genotypes somehow
+                    return;
+                }
+                final VariantContext match;
+                if (!matches.get(0).getSource().equals(vc.getSource())) {
```

Comment:
Can you put a note somewhere that we expect the input VCFs to NOT have split multi-alleleics?  That would destroy some of this logic.